src/java/org/apache/cassandra/journal/ActiveSegment.java

```diff
@@ -261,21 +264,35 @@ boolean shouldFlush()
      * TODO FIXME: calls from outside Flusher + callbacks
      * @return last synced offset
      */
-    synchronized int flush()
+    synchronized int flush(boolean fsync)
     {
         int allocatePosition = this.allocatePosition.get();
         if (lastFlushedOffset >= allocatePosition)
             return lastFlushedOffset;
 
         waitForModifications();
-        flushInternal();
+        if (fsync) flushInternal();
         lastFlushedOffset = allocatePosition;
```

Comment:
Nit: move this one line up and put the `flushIntearnal()` call (`fsyncInternal()`?) and updating the last fsync offset in one `if (fsync)` block?